doc/tapscript_opcodes.md

```diff
@@ -12,82 +12,72 @@ Taproot already increases a lot of resource limitations from segwitv0, so there
+         - If the scriptPubKey is a native segwit program, push the witness program(2-40) followed by a push for segwit version(0-1).
+      1. Define `OP_SUCCESS203` as `OP_INSPECTINPUTSEQUENCE`: Pop a `CScriptNum` input index `idx` and push the `nSequence`(4) as little-endian number.
+      1. Define `OP_SUCCESS204` as `OP_INSPECTINPUTISSUANCE`: Pop a `CScriptNum` input index `idx` and push the assetIssuance information if the asset has issuance, otherwise push an empty vector. Asset Issuance information is pushed as follows
+         - Push `nInflationKeys` as tuple, value(8, 32) followed by push for prefix(1). In case `nInflationKeys` is null, push an empty vector on stack top.
+         - Push `nAmount` as a tuple, value(8, 32) followed by a push for prefix(1). This cannot be null
+         - Push 32 byte `assetEntropy`
+         - Push 32 byte `assetBlindingNonce`
+   - Define `OP_SUCCESS205` as `OP_PUSHCURRENTINPUTINDEX` that pushes the current input index as `CScriptNum`. This can be used in conjunction with input introspection opcodes for inspecting current input.
+   - Output introspection opcodes:
+      1. Define `OP_SUCCESS206` as `OP_INSPECTOUTPUTASSET`: Pop a `CScriptNum` input index `idx` and push the `nAsset` as a tuple, first push the assetID(32), followed by the prefix(1)
+      1. Define `OP_SUCCESS207` as `OP_INSPECTOUTPUTVALUE`: Pop a `CScriptNum` input index `idx` and push the `nValue` as a tuple, value(8, 32) followed by prefix
+      1. Define `OP_SUCCESS208` as `OP_INSPECTOUTPUTNONCE`: Pop a `CScriptNum` input index `idx` and push the `nNonce`(33) onto the stack. If the nonce is null, push an empty vector onto the stack.
+      1. Define `OP_SUCCESS209` as `OP_INSPECTOUTPUTSCRIPTPUBKEY`: Pop a `CScriptNum` input index `idx` and push the scriptPubkey onto the stack.
+         - If the scriptPubKey is not a native segwit program, fail if the size of scriptPubKey is more than 520(`MAX_SCRIPT_ELEMENT_SIZE`) bytes, otherwise push the scriptPubKey on stack top. Next, push a `CScriptNum(-1)` to indicate a non-native segwit scriptPubKey.
+         - If the scriptPubKey is a native segwit program, push the witness program(2-40) followed by a push for segwit version(0-1).
+   - Transaction introspection opcodes:
+      1. Define `OP_SUCCESS210` as `OP_INSPECTVERSION`: Push the nVersion(4) as little-endian.
+      1. Define `OP_SUCCESS211` as `OP_INSPECTLOCKTIME`: Push the nLockTime(4) as little-endian.
+      1. Define `OP_SUCCESS212` as `OP_INSPECTNUMINPUTS`: Push the number of inputs(4) as little-endian
+      1. Define `OP_SUCCESS213` as `OP_INSPECTNUMOUTPUTS`: Push the number of outputs(4) as little-endian
+      1. Define `OP_SUCCESS214` as `OP_TXWEIGHT`: Push the transaction weight (4) as little-endian
```

Comment:
Consider making TXWEIGHT a 64 bit value.  I think it will commonly be used in fee-rate calculations.  The other case is some sort of absolute bound, but that mostly works as a 64 bit value too.

I could go either way on this though.